File: package/main/src/Math/linearCongruentialGenerator.ts
/**
 * Linear Congruential Generator for random number generation
 * @param seed Initial seed value
 * @param max Maximum value (default: 4294967296)
 * @param multiplier Multiplier parameter (default: 1664525)
 * @param increment Increment parameter (default: 1013904223)
 * @returns Random number generator function
 * @example
 * const lcg = linearCongruentialGenerator(12345);
 */
export const linearCongruentialGenerator = (
  seed: number,
  max = 4_294_967_296,
  multiplier = 1_664_525,
  increment = 1_013_904_223,
) => (multiplier * seed + increment) % max;

File: package/umt_python/src/math/linear_congruential_generator.py
def linear_congruential_generator(
    seed: int,
    max_value: int = 4_294_967_296,
    multiplier: int = 1_664_525,
    increment: int = 1_013_904_223,
) -> int:
    """
    Linear Congruential Generator for random number generation.

    Args:
        seed: Initial seed value.
        max_value: Maximum value (default: 4294967296).
        multiplier: Multiplier parameter (default: 1664525).
        increment: Increment parameter (default: 1013904223).

    Returns:
        Generated random number.

    Example:
        >>> linear_congruential_generator(12345)
        1406932606
    """
    return (multiplier * seed + increment) % max_value
